hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java

```diff
@@ -699,9 +693,9 @@ private static String escapeSpecialValueIfNeeded(String value) {
         if (value.isEmpty()) {
             // Escape empty String to INDEX_SYM_EMPTY (char `\u0002`)
             value = INDEX_SYM_EMPTY;
-        } else if (value == INDEX_VALUE_EMPTY) {
+        } else if (INDEX_VALUE_EMPTY.equals(value)) {
             value = "";
-        } else if (value == INDEX_VALUE_NULL) {
+        } else if (INDEX_VALUE_NULL.equals(value)) {
```

Comment:
these lines don't want to compare string chars, see the comment with the variable